test/common/event/scaled_range_timer_manager_test.cc

```diff
@@ -0,0 +1,450 @@
+  // should be a no-op, and if not (because a timer was fired), that's a problem that will be caught
+  // by the strict mock callback.
+  simTime().advanceTimeAsync(std::chrono::seconds(10));
+  dispatcher_.run(Dispatcher::RunType::Block);
+}
+
+TEST_F(ScaledRangeTimerManagerTest, DisableWhileDisabled) {
+  ScaledRangeTimerManager manager(dispatcher_, 1.0);
+
+  StrictMock<MockFunction<TimerCb>> callback;
+  auto timer = manager.createTimer(callback.AsStdFunction());
+
+  EXPECT_FALSE(timer->enabled());
+  timer->disableTimer();
+
+  EXPECT_FALSE(timer->enabled());
+}
+
+TEST_F(ScaledRangeTimerManagerTest, DisableWhilePending) {
+  ScaledRangeTimerManager manager(dispatcher_, 1.0);
+
+  StrictMock<MockFunction<TimerCb>> callback;
+  auto timer = manager.createTimer(callback.AsStdFunction());
+  timer->enableTimer(std::chrono::seconds(10), std::chrono::seconds(100));
+  EXPECT_TRUE(timer->enabled());
+
+  timer->disableTimer();
+  EXPECT_FALSE(timer->enabled());
+}
+
+TEST_F(ScaledRangeTimerManagerTest, DisableWhileActive) {
+  ScaledRangeTimerManager manager(dispatcher_, 1.0);
+
+  StrictMock<MockFunction<TimerCb>> callback;
+  auto timer = manager.createTimer(callback.AsStdFunction());
+
+  timer->enableTimer(std::chrono::seconds(5), std::chrono::seconds(100));
+
+  simTime().advanceTimeAsync(std::chrono::seconds(5));
+  dispatcher_.run(Dispatcher::RunType::Block);
+
+  EXPECT_TRUE(timer->enabled());
+
+  timer->disableTimer();
+  EXPECT_FALSE(timer->enabled());
+
+  // Run the dispatcher to make sure nothing happens when it's not supposed to.
+  simTime().advanceTimeAsync(std::chrono::seconds(100));
+  dispatcher_.run(Dispatcher::RunType::Block);
+}
+
+TEST_F(ScaledRangeTimerManagerTest, DisableFrontActiveTimer) {
+  ScaledRangeTimerManager manager(dispatcher_, 1.0);
+
+  StrictMock<MockFunction<TimerCb>> callback1, callback2;
+  EXPECT_CALL(callback2, Call);
+  auto timer1 = manager.createTimer(callback1.AsStdFunction());
+  auto timer2 = manager.createTimer(callback2.AsStdFunction());
+
+  // These timers have the same max-min.
+  timer1->enableTimer(std::chrono::seconds(5), std::chrono::seconds(30));
+  timer2->enableTimer(std::chrono::seconds(10), std::chrono::seconds(35));
+
+  simTime().advanceTimeAsync(std::chrono::seconds(5));
+  dispatcher_.run(Dispatcher::RunType::Block);
+  simTime().advanceTimeAsync(std::chrono::seconds(5));
+  dispatcher_.run(Dispatcher::RunType::Block);
+
+  timer1->disableTimer();
+  EXPECT_FALSE(timer1->enabled());
+  ASSERT_TRUE(timer2->enabled());
+
+  // After the original windows for both timers have long expired, only the enabled one should fire.
+  simTime().advanceTimeAsync(std::chrono::seconds(100));
+  dispatcher_.run(Dispatcher::RunType::Block);
+}
+
+TEST_F(ScaledRangeTimerManagerTest, DisableLaterActiveTimer) {
+  ScaledRangeTimerManager manager(dispatcher_, 1.0);
+
+  StrictMock<MockFunction<TimerCb>> callback1, callback2;
+  EXPECT_CALL(callback1, Call);
```

Comment:
Move this expect call just below this comment.  There's currently a big disconnect between this expectation and comments that refer to it.

// After the original windows for both timers have long expired, only the enabled one should fire.